src/main/java/org/candlepin/pki/impl/JSSPKIUtility.java

```diff
@@ -273,14 +401,15 @@ private CertificateExtensions buildStandardExtensions(CertificateExtensions cert
      *
      * A KeyIdentifier is a SHA-1 digest of the subjectPublicKey bit string from the ASN.1 above.
      *
-     * @param key the RSAPublicKey to use
+     * @param key the public key to use
      * @return an AuthorityKeyIdentifierExtension based on the key
      * @throws IOException if we can't construct a MessageDigest object.
      */
-    public static AuthorityKeyIdentifierExtension buildAuthorityKeyIdentifier(RSAPublicKey key)
+    public static AuthorityKeyIdentifierExtension buildAuthorityKeyIdentifier(PublicKey key)
         throws IOException {
         try {
-            MessageDigest d = MessageDigest.getInstance("SHA-1");
+            Provider provider = JSSProviderLoader.getProvider(true);
+            MessageDigest d = MessageDigest.getInstance("SHA-1", provider);
```

Comment:
I saw that, too. I didn't touch it because it's out of scope and requires its own investigation to determine whether or not it's safe to change here. If we don't already have a task for it, it's probably good to create one.

Reply:
Added a card in the backlog for this.